Plugins/DeleteUnusedBranches/DataGridViewCheckBoxHeaderCell.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Drawing;
+using System.Windows.Forms;
+using System.Windows.Forms.VisualStyles;
+
+namespace DeleteUnusedBranches
+{
+    public class DataGridViewCheckBoxHeaderCell : DataGridViewColumnHeaderCell
+    {
+        private Point _checkBoxLocation;
+        private Size _checkBoxSize;
+        private Point _cellLocation = new Point();
+        private CheckBoxState _cbState = CheckBoxState.UncheckedNormal;
+        private bool _checked = false;
+
+        public bool Checked
+        {
+            get { return _checked; }
+            set
+            {
+                _checked = value;
```

Comment:
you only want to do this if `_checked != value`, otherwise you will be invalidating the grid, which may be expensive.
https://github.com/gitextensions/gitextensions/pull/5100#discussion_r197610468